app/controllers/application_controller.rb

```diff
@@ -101,7 +101,7 @@ def sms_acct_creation_notice(resource_name, sms_status)
 
   def store_referring_location
     if request.referer && !request.referer.end_with?("users/sign_in")
-      session[:return_to] = request.referer
+      session[:return_to] = request.referer unless params[:ignore_referer]
```

Comment:
I think the unless condition probably belongs in the if statement.

In the future I think the `users/sign_in` should have that same param rather than using a check for the URL.

Reply:
good catch.